### DISCUSSION TITLE
Getting connection busy issue

### DISCUSSION THREAD

**Discussion by Buvi1234** [sqlalchemy/sqlalchemy#11183] [Usage Questions]
### Describe the use case

I am using sqlalchemy version 1.4.15, and i have created a orm session with the help socped_session registry and i have kept the connection pool size has 20. i have a web application with 4 parallel web request running.

Observation is in the connection pool the current checked out connection is 1, and with parallel request running the current checked out connection is still remains 1 and it does not increase on demand of the web application.

Am i missing anything here, could just help me on this regard.

### Databases / Backends / Drivers targeted

mssql : odbc driver 18
pyodbc: 4.0.35
python: 3.10.9
sqlalchemy: 1.4.51
Zope: 5.8.3

### Example Use

from sqlalchemy import create_engine
from sqlachemy.orm import sessionmaker, scoped_session

engine_option: {pool_size: 20}

engine = create_engine("username:password@dsn")

Dsn based connection

Session = scoped_session(sessionmaker(bind=engine))
session = Session() --> thread local session

if i say session.bind.pool.status() --> the current out connection is 1.

**Comment by zzzeek** [comment]
hi -

this sounds like something is not right with your application server config overall.   Code looks fine and there's nothing here to indicate SQLAlchemy's connection pool is any kind of bottleneck.  I would consult with Zope discussion groups where they should be pretty helpful in helping you to diagnose your issue.

**Reply by Buvi1234** [reply]
@zzzeek, on what use cases the current checked out connection will remain one, if we have multiple request incoming.

Instead of instantiating the scoped registry like session = Session(), can i use the scoped_session class directly.

**Reply by CaselIT** [reply]
your example in not clear at all.
is this line `session = Session()` at module level?

if so that's not correct. You need to call `Session()` at the beginning of each request, when call its close at the end of the request

**Reply by Buvi1234** [reply]
No, session = session() is at the global level, i am creating a dictionary and storing the key as some db session(name) and the value has session object, on call of every request getting the value from the dictionary(session which is created once with session factory instance) and I am attaching it to the request, and connection pool is set to 20. But the current checked out connection remains 1. For subsequent request gets connection busy issue

**Reply by CaselIT** [reply]
Yes, because you should not do session = session() at the global level.

**Reply by Buvi1234** [reply]
Thank you, for the suggestion. The session() will have a connection pool(Queuepool). If the connection is busy it should take another connection which is idle in the connection pool right but I am seeing the checked out connection remains 1

**Reply by CaselIT** [reply]
A single session instance, that you are creating with `session = session()`, doesn't do that.

**Reply by Buvi1234** [reply]
So, shall directly use the scoped session class directly?, in the earlier version of sqlalchemy it was working fine.

**Reply by CaselIT** [reply]
As far as I know the behaviour of sqlalchemy has never changed here, so it should behave like that in any sqlalchemy version

**Reply by Buvi1234** [reply]
thank you for the help
